src/ai/backend/agent/docker/kernel.py

```diff
@@ -127,6 +132,52 @@ async def get_service_apps(self):
         result = await self.runner.feed_service_apps()
         return result
 
+    async def check_commit_tag(self, commit_path: Path, get_lock: bool = False):
+        container_id: str = str(self.data["container_id"])
+        tag_path = commit_path / "tags" / container_id
+
+        tag_path.parent.mkdir(parents=True, exist_ok=True)
+        try:
+            with open(tag_path, mode="x") as f:
+                if get_lock:
+                    f.write("")
+        except FileExistsError:
+            return int(False)
+        return int(True)
+
+    async def commit(self, image_commit_path: Path, path: str):
+        assert self.runner is not None
+
+        container_id: str = str(self.data["container_id"])
+
+        filepath = image_commit_path / path
+        filename = Path(filepath).name
+
+        try:
+            Path(filepath).parent.mkdir(exist_ok=True, parents=True)
+        except ValueError:  # parent_path does not start with work_dir!
+            raise AssertionError("malformed committed path.")
+
+        loop = current_loop()
+        async with closing_async(Docker()) as docker:
+            async with docker._query(
+                f"containers/{container_id}/export",
+                method="GET",
+            ) as response:
+
+                data: bytes = await response.read()
```

Comment:
Blank lines should be minimized unless it is better to group statements as separate paragraphs. (It's just like writing.)